### ISSUE TITLE
Testing notification of custom LWRPs does not behave correctly

### ISSUE THREAD

**Issue by ghost** [sethvargo/chefspec#546]
Imaging a simple LWRP **test**.

``` ruby
actions :create
attribute :name, name_attribute: true
```

``` ruby
action(:create) do
  new_resource.updated_by_last_action(new_resource.name == 'bar')
end
```

This means, that only if I create the resource **test[bar]**, it fires its notifications.

``` ruby

test 'bar' do
  action :create
  notifies :restart, 'service[name]'
end

```

**test[foo]**, shouldn't fire a notification.

``` ruby

test 'foo' do
  action :create
  notifies :restart, 'service[name]'
end
```

The appropriate tests could look like this:

``` ruby
describe 'test::default' do

  cached(:chef_run) {
    ChefSpec::ServerRunner.new.converge(described_recipe)
  }


  it 'should notify service[name] when test is called bar' do
    bar = chef_run.test('bar')
    expect(bar).to notify('service[name]').to(:restart)
  end

  it 'should NOT notify service[name] when test is called foo' do
    foo = chef_run.test('foo')
    expect(foo).to_not notify('service[name]').to(:restart)
  end

end

```

And the matcher

``` ruby
ChefSpec.define_matcher :test

def test(resource_name)
  ChefSpec::Matchers::ResourceMatcher.new(:test, :create, resource_name)
end
```

Now, when you run this test, it's saying:

```
     Failure/Error: expect(foo).to_not notify('service[name]').to(:restart)
       expected "test[foo]" to not notify "service[name]", but it did.
```

**Comment by dblessing** [comment]
ChefSpec is testing the catalog. So, it is correct to say each of those resources *does* cause a notification to be set up in the catalog. You need an integration test to determine if a specific event causes a notification to actually be sent. This is my understanding of what you're describing here.

**Comment by yves-vogl** [comment]
@sethvargo Could you please reopen this issue now?

**Comment by sethvargo** [comment]
Please see [my response on your PR](https://github.com/sethvargo/chefspec/pull/547#issuecomment-68649062)

**Comment by sethvargo** [comment]
@yves-vogl also, @dblessing gave a much shorter explanation than mine, which is correct.

**Comment by till** [comment]
@sethvargo I spent a good amount of time on the same problem before I found this Github issue.

I totally get the explanation, so thanks for that.

However, I am wondering if since ChefSpec will always assume that resources notify, maybe something like `expect(resource).not_to notify('something[else]')` should be added to the documentation so people do not try this and instead do an integration test to start with?

Maybe I am not sure where to draw the line between ChefSpec and _actual_ integration tests.
